ikarus/finiteelements/mechanics/materials/muesli/mueslifinite.hh

```diff
@@ -0,0 +1,161 @@
+// SPDX-FileCopyrightText: 2021-2024 The Ikarus Developers mueller@ibb.uni-stuttgart.de
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+/**
+ * \file Muesli.hh
+ * \brief Implementation of the Muesli material model.
+ * \ingroup materials
+ */
+
+#pragma once
+
+#include <muesli/muesli.h>
+
+#include <Eigen/Eigen>
+
+#include <ikarus/finiteelements/mechanics/materials/interface.hh>
+#include <ikarus/finiteelements/mechanics/materials/muesli/mueslihelpers.hh>
+#include <ikarus/utils/tensorutils.hh>
+
+namespace Ikarus::Materials::Muesli {
+
+/**
+ * \brief Wrapper class for finite strain materials from the muesli library. It can be templated with all materials
+ * derived from muesli::finiteStrainMaterial. It models the Ikarus material interface.
+ *
+ * \tparam FM muesli material model implementation
+ * \remark Please cite \cite portillo_muesli_2017 if you use any materials from the muesli library
```

Comment:
For this to work I think we need a global variable to be set. 
I just had the idea to have global ikarus parameters, that are initialized through `init()` and can then be accessed throghout the code through a static singleton object.  But I think this has to be discussed in a seperate PR

Reply:
Yes. I am not exactly sure how the implementation will look like atm but yes it can also be done separately in a different PR. Can you note it down in some issue?